great_ape_safe/ape_api/ebtc.py

```diff
@@ -866,3 +872,211 @@ def authority_set_authority(self, new_authority):
                 EmptyBytes32,
                 delay + 1,
             )
+
+    #### ===== CDP OPS ===== ####
+
+    def _assert_collateral_balance(self, coll_amount):
+        total_coll_bal = self.collateral.balanceOf(self.safe.address)
+        assert total_coll_bal >= self.collateral.getSharesByPooledEth(coll_amount)
```

Comment:
If you take a look at the stETH [contract](https://etherscan.io/address/0x17144556fd3424edc8fc8a4c940b2d04936d17eb#code#F27#L300), the `balanceOf` function returns the amount in terms of ETH balance whereas the `getSharesByPooledEth` returns the equivalent amount shares for a given ETH amount. This means that this comparison isn't really comparing apples to apples and with a non 1 PPFS it would break for an exact balance deposit.

Given that your functions seems to be taking the stETH in terms of token balance (to then use to estimate the debt amount to borrow based on the spot price), I think the best here is to compare the input amount to the held balance and that's it.

ie.:
```
assert total_coll_bal >= coll_amount
```

Reply:
the situation here is that the collateral token in sepolia differs from the logic in the `CollateralTokenTester` that the balance of for the user when using `forceDeposit` seems to be using internally [getSharesByPooledEth](https://sepolia.etherscan.io/address/0x97BA9AA7B7DC74f7a74864A62c4fF93b2b22f015#code#L290), then to compare the right unit in the assert its seems to me that the conversion here is necessarily, at least, for testnet

lmk if im missing something, but try to do `assert total_coll_bal >= coll_amount` locally and likely reverts will trigger since will not be comparing apples to apples in this particular case

i agree that for production the logic differs

Reply:
Ah, interesting, good catch. Seems like it was also noticed by the Solidity team and is being addressed in this [PR](https://github.com/ebtc-protocol/ebtc/pull/739) on the eBTC project.

I suggest then that we at least leave a note to this in here. There will be a another Sepolia deployment coming out soon with these patched, we should modify the functions when those are wired.

Reply:
included hints for reminder in 9e843c88de21d5a134f48a8440a9a8386f6890a2